src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -250,16 +250,31 @@ public function getBoolean($key, $default = false, $deep = false)
      *
      * @param string $key     Key.
      * @param mixed  $default Default = null.
-     * @param bool   $deep    Default = false.
      * @param int    $filter  FILTER_* constant.
      * @param mixed  $options Filter options.
+     * @param bool   $deep    Default = false.
      *
      * @see http://php.net/manual/en/function.filter-var.php
      *
      * @return mixed
      */
-    public function filter($key, $default = null, $deep = false, $filter = FILTER_DEFAULT, $options = array())
+    public function filter($key, $default = null, $filter = FILTER_DEFAULT, $options = array(), $deep = false)
     {
+        static $filters = null;
+
+        if (null === $filters) {
+            foreach (filter_list() as $tmp) {
+                $filters[filter_id($tmp)] = 1;
+            }
+        }
+        if (is_bool($filter) || !isset($filters[$filter]) || is_array($deep)) {
+            @trigger_error('Passing the $deep boolean as 3rd argument of the '.__METHOD__.' method is deprecated since version 2.8 and will be removed in 3.0.', E_USER_ERROR);
```

Comment:
to pass it as last argument is also deprecated. this can be important when we add new arguments later.